eos/constraint.cc

```diff
@@ -1145,19 +1146,30 @@ namespace eos
                     throw InternalError("make_multivariate_gaussian_covariance_constraint<measurements=" + stringify(dim_meas) + ",predictions=" + stringify(dim_pred) + ">: " + name.str() + ": '" + this->observables[i].str() + "' is not a valid observable name");
             }
 
-            // create GSL vector for the mean
-            gsl_vector * means = gsl_vector_alloc(dim_meas);
-            gsl_vector_memcpy(means, this->means);
+	    //If specified, these options allow to specify a subset of the measurements
+	    unsigned begin = destringify<unsigned>(options.get("begin", "0"));
+	    unsigned end = destringify<unsigned>(options.get("end", stringify(dim_meas)));
+	    unsigned sub_dim = end - begin;
+
+	    if (! sub_dim>0)
```

Comment:
I would make it clearer and write ```0 > sub_dim```, unless you have a reason for this I do not see.

Reply:
I have no reason :)

Reply:
Hmm all well thought this was stupid. I defined sub_dim as unsigned, so:
 1 - The if statement is always true
 2 - If `end - begin < 0`, sub_dim is not well defined and will certainly lead to an error...

Reply:
Yes, good point. I didn't see that. I would suggest to use signed integeers, and
do the error checking more explicitly:

```
if (end > dim_meas)
    throw OptionValueError(... + "cannot use a value of 'end' pointing beyond the number of measurements.");

if (begin < 0)
   throw OptionValueError(... + "cannot use a negative value for 'begin'");

if (begin >= end)
   throw OptionValueError(... + "cannot use a value for 'begin' equal to or larger than 'end'");
```

No need to revert the PR, just push a correction in a separate commit on top of this branch and we handle it here.

Reply:
Ok I do this, thank you